frontend/src/components/beta/accounts/AccountFungibleTokenHistory.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from "react";
+import { shortenString } from "../../../libraries/formatting";
+
+import { styled } from "../../../libraries/styles";
+import {
+  AccountFungibleToken,
+  AccountFungibleTokenHistoryElement,
+} from "../../../types/common";
+import { trpc } from "../../../libraries/trpc";
+import moment from "moment";
+import { useTranslation } from "react-i18next";
+import { TokenAmount } from "../../utils/TokenAmount";
+import LinkWrapper from "../../utils/Link";
+import { buildAccountUrl } from "../../../hooks/use-account-page-options";
+
+const TableWrapper = styled("div", {
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "center",
+  fontFamily: "Manrope",
+  paddingHorizontal: 24,
+  borderRadius: 8,
+  border: "1px solid #e8e8e8",
+  marginLeft: 20,
+  flex: 2,
+
+  "@media (max-width: 768px)": {
+    marginLeft: 0,
+    marginTop: 20,
+  },
+});
+
+const TableHeader = styled("thead", {
+  textTransform: "uppercase",
+  color: "#c4c4c4",
+  borderBottom: "1px solid #e8e8e8",
+
+  fontSize: 12,
+  fontWeight: 600,
+});
+
+const TableHeaderCell = styled("th", {
+  paddingVertical: 20,
+});
+
+const TableRow = styled("tr", {
+  fontSize: 14,
+  fontWeight: 500,
+  height: 50,
+
+  "& + &": {
+    borderTop: "1px solid #e8e8e8",
+  },
+});
+
+const TableElement = styled("td", {
+  verticalAlign: "top",
+  paddingVertical: 8,
+  paddingRight: 12,
+});
+
+const Link = styled(LinkWrapper, {
+  cursor: "pointer",
+
+  variants: {
+    disabled: {
+      true: {
+        cursor: "inherit",
+      },
+    },
+  },
+});
+
+type ItemProps = {
+  item: AccountFungibleTokenHistoryElement;
+  token: AccountFungibleToken;
+};
+
+const AccountFungibleTokenHistoryElementView: React.FC<ItemProps> = React.memo(
+  ({ item, token }) => {
+    const { t } = useTranslation();
+    return (
+      <TableRow>
+        <Link
+          href={
+            item.counterparty
+              ? buildAccountUrl({
+                  accountId: item.counterparty,
+                  tab: "fungible-tokens",
+                })
+              : ""
+          }
+          shallow
+          disabled={!item.counterparty}
+        >
+          <TableElement>
+            {item.counterparty
+              ? shortenString(item.counterparty)
+              : item.direction === "in"
+              ? "MINT"
+              : "BURN"}
+          </TableElement>
+        </Link>
+        <TableElement>
+          <TokenAmount
+            token={{ ...token, balance: item.amount }}
+            sign={item.direction === "out" ? -1 : 1}
```

Comment:
This looks odd to me to use -1 / 1 as part of the component interface, to be honest. Why don't just use `balance` sign for that?

Reply:
I think we can add a check for negative numbers in `formatToPowerOfTen`, that will work.
We still need some specific prop to show "+" there, I chose a "prefix".